File: Coding Problems - Python/Arrays/Two_sum.py
nums = [3, 5, -4, 8, 11, 1, -1, 6]
target = 10
prev_hashmap = {}

for n in nums:
    k = target - n
            
    if k in prev_hashmap:
        print([k, n])
    else:
        prev_hashmap[n] = True

File: Coding_Problems_Java/Two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hmap = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int K = target - nums[i];

            if (hmap.containsKey(K)){
                return new int []{hmap.get(K), i};
            }
            else{
                hmap.put(nums[i],i);
            }
        }
        return new int []{};   
    }
}
